### ISSUE TITLE
Searching by folder and/or tag

### ISSUE THREAD

**Issue by faouzzz** [keeweb/keeweb#1588]
**Is your feature request related to a problem? Please describe.**
I have many same account name on differents folders in my kbdx. when i search i have too much result. i need to add folder name or tag in the search field to reduce then number of result.

**Describe the solution you'd like**
i think a search like **folder/search_term** or **#tag/search_term** should set on top of result the items found in **folder** or tagged with **tag** ?

**Describe alternatives you've considered**
Scrolling and completing the search manually...

**Additional context**

**Comment by antelle** [comment]
Hi! I agree that it would be nice to have a tag input in the search field, however I don't think it should sort anything, otherwise the purpose of two fields is lost:
<img width="243" alt="fields" src="https://user-images.githubusercontent.com/633557/91298253-f8e82080-e79f-11ea-9aee-f8e13fdc77f2.png">
What it however can do, is filter the search results by given tag/folder. It's better to highlight a tag in the search field to avoid confusion. Something like this ([fiddle](https://jsfiddle.net/c6Lobjkf/1/)):
<img width="290" alt="Screen Shot 2020-08-26 at 14 02 39" src="https://user-images.githubusercontent.com/633557/91301215-d3a9e100-e7a4-11ea-9bf9-119df9bce1e4.png">

Text should transform into highlighted versions on typing `<foo>/`, after pressing <kbd>/</kbd>, and `#foo␣`, after a space or enter. If a backspace is pressed immediately after auto-replacement, it should revert to the text version. Pressing backspace or delete on this special entity should delete it all at once.

This is at least how I see it to be implemented, however I would expect someone willing to do it to contribute, since it's quite a lot of work here.

**Comment by BrianLeishman** [comment]
Just throwing a use case in, coming from a kdbx file that was only used with the official client, that had all the passwords organized by folder names instead of titles. keepass's search would include the folder names as things searched against, but searching the same kdbx file in KeeWeb becomes impossible

I would expect like a folder option here, maybe not default checked to keep people's search behavior from changing on them
![image](https://user-images.githubusercontent.com/1208702/116290105-1d4a3180-a761-11eb-83e6-c22eb2e47873.png)

Like this example, I have a few credentials in the "Asana" folder but nothing comes up when searched 
![image](https://user-images.githubusercontent.com/1208702/116290265-4ec2fd00-a761-11eb-8c5c-dee011a33995.png)

**Comment by antelle** [comment]
@BrianLeishman that's a different thing, searching groups is #152

**Comment by BrianLeishman** [comment]
Yup I just came to the same issue, I totally misread the original question, I also didn't click on https://github.com/keeweb/keeweb/issues/152 because it was titled about sorting, and just skimmed this question too quickly

**Comment by epick** [comment]
Any news about this feature? I have the same problem I can't use the search box to find specific entries inside a Client folder XX, I know there are a lot of people who need to find results by folders and not by entries,

Thanks
